src/main/java/org/mockito/internal/stubbing/defaultanswers/ReturnsSmartNulls.java

```diff
@@ -89,9 +92,27 @@ public Object answer(InvocationOnMock currentInvocation) throws Throwable {
             if (isToStringMethod(currentInvocation.getMethod())) {
                 return "SmartNull returned by this unstubbed method call on a mock:\n"
                         + unstubbedInvocation;
+            } else if (isMethodOf(MockAccess.class, currentInvocation.getMock(), currentInvocation.getMethod())) {
```

Comment:
Fixing `ReturnsSmartNullsTest` test cases, @TimvdLippe this is a really interesting one, the `ReturnsSmartNulls` essentially creates mock for a mock, this is fine by and large but mocks could implement `MockAccess` (Mockito's internal), and the methods of  `MockAccess` are also stubbed with default answer, failing the tests (no suitable answer exists).

The fix is to call the real methods of the `MockAccess`, the tests are green but I am not entirely sure this is the right place, may be `MockHandlerImpl` could be the one, would appreciate some thoughts here.

Thank you.

Reply:
I think I prefer to have the code here, as our `MockHandlerImpl` is in the hot path and we should keep it as clean and fast as possible. So I would be okay with this.